tests/python/test_multidevice.py

```diff
@@ -1688,3 +1688,78 @@ def test_transformer_backward(multidevice_test, benchmark):
     _assert_shape_dtype(inp_grad, [b, s, e], torch.bfloat16)
 
     benchmark.pedantic(benchmark_fn, rounds=5)
+
+
+class OverlapAGMatmulStreamOutermost(FusionDefinition):
+    def __init__(self, M, K, N, S, num_devices):
+        super().__init__()
+        self._M = M
+        self._K = K
+        self._N = N
+        self._S = S
+        self._num_devices = num_devices
+        self.use_multidevice_executor()
+
+    def definition(self) -> None:
+        M, K, N, S, D = (
+            self._M,
+            self._K,
+            self._N,
+            self._S,
+            self._num_devices,
+        )
+        self.x = self.define_tensor(
+            shape=[S, D, M // (S * D), K], contiguity=True, dtype=DataType.BFloat16
+        )
+        self.weight = self.define_tensor(
+            shape=[N, K], contiguity=True, dtype=DataType.BFloat16
+        )
+        self.bias = self.define_tensor(
+            shape=[N], contiguity=True, dtype=DataType.BFloat16
+        )
+
+        self.out = self.ops.linear(
+            self.x, self.weight, self.bias
+        )  # [S, D, M//(S*D), N]
+
+        self.add_output(self.out)
+
+    def multidevice_schedule(self):
+        mesh = nvfuser.DeviceMesh(range(self._num_devices))
+        for tv in [
+            self.x,
+            self.weight,
+            self.bias,
+            self.out,
+        ]:
+            self.sched._set_device_mesh(tv, mesh)
+
+        self.sched.parallelize(self.x, 1, nvfuser.ParallelType.mesh_x)
+        self.sched.parallelize(self.out, 0, nvfuser.ParallelType.stream)
+
+
+@pytest.mark.mpi
+def test_overlap_allgather_matmul_stream_outermost(multidevice_test, benchmark):
+    M, K, N, S, D = 256, 128, 64, 8, multidevice_test.size
```

Comment:
```suggestion
    m, k, n, s, d = 256, 128, 64, 8, multidevice_test.size
```
just to be consistent with the style in existing Python tests and https://peps.python.org/pep-0008/#function-and-variable-names. (I'm a bit surprised Lint didn't complain about this)

Reply:
No problem to use lower letters, but for my own understanding, why don't you apply the capital letters described for constants here https://peps.python.org/pep-0008/#constants?

Reply:
`d` is not a constant. For other sizes, I suspect at some point we'll want to make them parameters so I treat them variables at this moment, but I'm happy with those go either way as long as they are consistent within a test file and even better with other Python test files if possible.